crates/router/src/core/payments/helpers.rs

```diff
@@ -280,9 +280,13 @@ pub fn validate_request_amount_and_amount_to_capture(
 
 pub fn validate_mandate(
     req: impl Into<api::MandateValidationFields>,
-) -> RouterResult<Option<api::MandateTxnType>> {
+) -> CustomResult<Option<api::MandateTxnType>, errors::ApiErrorResponse> {
     let req: api::MandateValidationFields = req.into();
-    match req.is_mandate() {
+    match req
+        .is_mandate()
+        .change_context(errors::ApiErrorResponse::MandateValidationFailed {
+            reason: "Expected one out of mandate_id and mandate_data but got both".to_string(),
```

Comment:
can we have an CONST for this string as it used in one more place

Reply:
No since it is leveraging the error `MandateValidationFailed` it expects a reason which need not to be a constant here